winnowing.py

```diff
@@ -3,122 +3,123 @@
-    hashval = hashval.hexdigest()[-4 :]
-    hashval = int(hashval, 16)  #using last 16 bits of sha-1 digest
-    return hashval
+class winnowing():
+    def __init__(self,file1,file2):
+        self.file1=file1
+        self.file2=file2
+        self.token1 = tokenize(file1)  #from cleanUP.py
+        self.str1 = toText(self.token1)
+        self.token2 = tokenize(file2)
+        self.str2 = toText(self.token2)
+        self.kGrams1 = self.kgrams(self.str1)  #stores k-grams, their hash values and positions in cleaned up text
+        self.kGrams2 = self.kgrams(self.str2)
+        self.HL1 = self.hashList(self.kGrams1)  #hash list derived from k-grams list
+        self.HL2 = self.hashList(self.kGrams2)
+        self.fpList1 = self.fingerprints(self.HL1)
+        self.fpList2 = self.fingerprints(self.HL2)
 
-#function to form k-grams out of the cleaned up text
-def kgrams(text, k = 25):
-    tokenList = list(text)
-    n = len(tokenList)
-    kgrams = []
-    for i in range(n - k + 1):
-        kgram = ''.join(tokenList[i : i + k])
-        hv = hash(kgram)
-        kgrams.append((kgram, hv, i, i + k))  #k-gram, its hash value, starting and ending positions are stored
-        #these help in marking the plagiarized content in the original code.
-    return kgrams
-
-#function that returns the index at which minimum value of a given list (window) is located
-def minIndex(arr):
-    minI = 0
-    minV = arr[0]
-    n = len(arr)
-    for i in range(n):
-        if arr[i] < minV:
-            minV = arr[i]
-            minI = i
-    return minI
+    def plagiarismCheck(self):
+        file1 = self.file1
+        file2 = self.file2
+        f1 = open(file1, "r")
+        start = []   #to store the start values corresponding to matching fingerprints
+        end = []   #to store end values
+        code = f1.read()  #original code
+        newCode = ""   #code with marked plagiarized content
+        points = []
+        for i in self.fpList1:
+            for j in self.fpList2:
+                if i == j:   #fingerprints match
+                    flag = 0
+                    match = self.HL1.index(i)   #index of matching fingerprints in hash list, k-grams list
+                    newStart = self.kGrams1[match][2]   #start position of matched k-gram in cleaned up code
+                    newEnd = self.kGrams1[match][3]   #end position
+                    for k in self.token1:
+                        if k[2] == newStart:   #linking positions in cleaned up code to original code
+                            startx = k[1]
+                            flag = 1
+                        if k[2] == newEnd:
+                            endx = k[1]
+                    if flag == 1:
+                        points.append([startx, endx])
+        points.sort(key = lambda x: x[0])
+        points = points[1:]
+        mergedPoints = []
+        mergedPoints.append(points[0])
+        for i in range(1, len(points)):
```

Comment:
No need of doing range(len(points)). Get the elements in the list directly

Reply:
solved sir